rastervision/backend/pytorch_chip_classification.py

```diff
@@ -217,17 +218,59 @@ def train(self, tmp_dir):
         num_labels = len(databunch.label_names)
         model = get_model(
             self.train_opts.model_arch, num_labels, pretrained=True)
+        if torch.cuda.device_count() > 1:
+            model = nn.DataParallel(model)
         model = model.to(self.device)
         model_path = join(train_dir, 'model')
 
         # Load weights from a pretrained model.
+        def convert_state_dict_to_dataparallel(state_dict):
+            '''
+            This function converts the state dict of a model created without
+            the nn.DataParallel module to one that can be read by a nn.DataParallel
+            enabled model.
+
+            Args:
+                state_dict (ordered dict): state_dict of the model
+            '''
+            from collections import OrderedDict
+            new_state_dict = OrderedDict()
+            for key, value, in state_dict.items():
+                name = 'module.' + key  # add 'module.'
+                new_state_dict[name] = value
+            return new_state_dict
+
+        def convert_state_dict_from_dataparallel(state_dict):
+            '''
+            This function converts the state dict of a model created with
+            the nn.DataParallel module to one that can be read into a model
+            without the nn.DataParallel.
+
+            Args:
+                state_dict (ordered dict): state_dict of the model
+            '''
+            from collections import OrderedDict
+            new_state_dict = OrderedDict()
+            for key, value, in state_dict.items():
+                name = key[7:]  # remove 'module.'
+                new_state_dict[name] = value
+            return new_state_dict
+
         pretrained_uri = self.backend_opts.pretrained_uri
         if pretrained_uri:
             log.info('Loading weights from pretrained_uri: {}'.format(
                 pretrained_uri))
             pretrained_path = download_if_needed(pretrained_uri, tmp_dir)
-            model.load_state_dict(
-                torch.load(pretrained_path, map_location=self.device))
+            state_dict = torch.load(pretrained_path, map_location=self.device)
+            if 'module' in list(
```

Comment:
Even if you were to go with this approach, you would need to also modify the code in `load_model` since that also loads model weights.

Reply:
To implement what I was suggesting, I think it would be good to create a utility function to save model weights and then use that function wherever checkpoints are saved. The function would need to check if the model is of type `DataParallel` and if so, strip off the module key. Otherwise, just save as normal.

Reply:
Agreed, i'll get to it

Reply:
Done